src/joint_moteus.cpp

```diff
@@ -0,0 +1,294 @@
+      can_bus_(can_bus),
+      can_id_(can_id),
+      query_type_(kDefaultQuery),
+      moteus_kp_(moteus_kp),
+      moteus_kd_(moteus_kd) {}
+
+JointMoteus::JointMoteus(int can_id,
+                         int can_bus,
+                         int direction,
+                         float zero_offset,
+                         float gear_ratio,
+                         float max_torque,
+                         float pos_min,
+                         float pos_max,
+                         float soft_start_duration_ms,
+                         ::mjbots::moteus::QueryCommand kDefaultQuery,
+                         float moteus_kp,
+                         float moteus_kd)
+    : JointBase("", direction, zero_offset, gear_ratio, max_torque, pos_min, pos_max, soft_start_duration_ms),
+      can_bus_(can_bus),
+      can_id_(can_id),
+      query_type_(kDefaultQuery),
+      moteus_kp_(moteus_kp),
+      moteus_kd_(moteus_kd) {}
+
+JointMoteus::JointMoteus(MoteusJointConfig config)
+    : JointBase(config.name, 
+                config.direction,
+                config.zero_offset,
+                config.gear_ratio,
+                config.max_torque,
+                config.pos_min,
+                config.pos_max,
+                config.soft_start_duration_ms),
+        can_id_(config.can_id),
+        can_bus_(config.can_bus),
+        query_type_(config.query_type),
+        moteus_kp_(config.moteus_kp),
+        moteus_kd_(config.moteus_kd) {}
+        
+void JointMoteus::UpdateMoteus(::mjbots::moteus::QueryResult reply_message){
+                    UpdateState(2 * M_PI * reply_message.position, 2 * M_PI * reply_message.velocity, reply_message.torque);
+                    mode_ = reply_message.mode;
+                    q_current_ = reply_message.q_current;
+                    d_current_ = reply_message.d_current;
+                    voltage_ = reply_message.voltage;
+                    temperature_ = reply_message.temperature;
+                    fault_ = reply_message.fault;
+}
+
+bool JointMoteus::is_open_loop() {
+  bool rv;
+  if(query_type_.position == ::mjbots::moteus::Resolution::kIgnore ||
+  query_type_.velocity == ::mjbots::moteus::Resolution::kIgnore){
+    rv = true;
+  } else {
+    rv = false;
+  }
+  return rv;
+}
+
+int JointMoteus::get_can_id() const {
+  return can_id_;
+}
+
+int JointMoteus::get_can_bus() const {
+  return can_bus_;
+}
+
+const ::mjbots::moteus::QueryCommand JointMoteus::get_query_command() const {
+  return query_type_;
+}
+
+const ::mjbots::moteus::Mode & JointMoteus::get_mode_reference() const {
+  if(query_type_.mode == ::mjbots::moteus::Resolution::kIgnore){
+    LOG_WARN("Mode resolution is set to kIgnore, while attempting to log register");
+  }
+  return mode_;
+}
+
+const ::mjbots::moteus::Mode JointMoteus::get_mode() const {
+  if(query_type_.mode == ::mjbots::moteus::Resolution::kIgnore){
+    LOG_WARN("Mode resolution is set to kIgnore, while attempting to log register");
```

Comment:
Should also mention this is in the `QueryCommand`,  resolution is also set in the data sent to the moteus.